IMPLEMENTATION.md

```diff
@@ -0,0 +1,23 @@
+### How to determine the port number that AA listens to as a grpc service
+
+In the container image decryption operation of kata cc V0 architecture, ocicrypt-rs parses the configuration file`/etc/containerd/ocicrypt/ocicrypt_keyprovider.conf` to obtain the port number of the keyprovider, for example:
+
+```
+{
+        "key-providers": {
+                "attestation-agent": {
+                				"grpc": "127.0.0.1:$port"
+                }
+        }
+}
+```
+
+At present, the implementation of AA adopts the scheme of listening to the fixed port number 44444
+
+### gRPC or ttRPC
+
+At present, AA uses gRPC protocol to implement the server of keyprovider for ocicrypt-rs. Compared with gRPC, ttRPC has the advantage of lighter weight. AA, as a memory resident service on the client side of kata cc V0 architecture, using lightweight ttRPC will save more resources.
+
+### How can AA obtain the access information (such as IP address) of KBS?
+
+AA needs to initiate decryption request to Key Broker Service (KBS) at the remote end after receiving the image decryption request from ocicrypt-rs, and then perform HW-tee remote attachment. So where does the information about connecting KBS come from?
```

Comment:
AA should get the keyURL from ocicrypt-rs. the keyURL is carried in the image layer annotation.

Reply:
We have considered this approach before, but it will expose the access information of KBS and bring unnecessary security risks.